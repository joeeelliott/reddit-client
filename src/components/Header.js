import React from 'react';
import SideNav from './SideNav'; 
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { useSelector, useDispatch } from 'react-redux'; 
import { selectShowSideNav, showSideNav } from '../features/sideNavSlice';

const Header = () => {
  const sideNavState = useSelector(selectShowSideNav)
  const dispatch = useDispatch();

  const handleShowSideNavClick = (e) => {
    sideNavState.toggleSideNav ? e.currentTarget.classList.toggle('Header_search-icon-rotate') : e.currentTarget.classList.toggle('Header_search-icon-rotate'); 

    dispatch(showSideNav()); 
  }

  return (
    <div className="Header_header">
      <div className="Header_toggle-btn-container">
        <FontAwesomeIcon icon="search" data-test-Header_search-icon data-test-Header_font-awesome-icon onClick={handleShowSideNavClick} />
      </div>

      <div className="Header_logo-title-container">
        <FontAwesomeIcon icon={['fab', 'reddit']} className="Header_font-awesome-icon" />

        <h1 className="Header_title">Reddit<span className="Header_title-span">Sample</span></h1>
      </div>

      <SideNav />
    </div>
  )
}

export default Header;