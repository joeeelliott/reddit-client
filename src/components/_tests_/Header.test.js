import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import Header from '../Header'; 

const mockStore = configureStore([]); 

describe('My Connected React-Redux Component', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      toggleSideNav: false,
      sideNavIconClicked: false,
      sideNavHidden: true,
    });
  });

  component = renderer.create(
    <Provider store={store}>
      <Header />
    </Provider>
  );

  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
  // received the following error for the above test 
  //  FAIL  src/components/_tests_/Header.test.js
  // ‚óè Test suite failed to run
  // TypeError: Cannot read property 'getState' of undefined
});