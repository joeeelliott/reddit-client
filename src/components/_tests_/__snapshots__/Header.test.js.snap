// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `
<div
  className="Header_header"
>
  <div
    className="Header_toggle-btn-container"
  />
  <div
    className="Header_logo-title-container"
  >
    <h1
      className="Header_title"
    >
      Reddit
      <span
        className="Header_title-span"
      >
        Sample
      </span>
    </h1>
  </div>
  <div>
    <div
      className="SideNav_hide-nav"
    >
      <form>
        <label>
          Search: 
        </label>
        <input
          placeholder="Search term here..."
          type="text"
        />
        <label>
          Filter: 
        </label>
        <select
          id="filters"
          name="filters"
        >
          <option
            value=""
          />
          <option
            value=""
          />
          <option
            value=""
          />
          <option
            value=""
          />
        </select>
        <div
          className="SideNav_btn-container"
        >
          <button
            className="SideNav_btn"
            onClick={[Function]}
          >
            Confirm
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
